paper-ripple.html

```diff
@@ -116,362 +113,418 @@
+        this.waveContainer.style.top =
+          (this.containerMetrics.height - this.containerMetrics.size) / 2 + 'px';
+        this.waveContainer.style.left =
+          (this.containerMetrics.width - this.containerMetrics.size) / 2 + 'px';
+
+        this.waveContainer.style.width = this.containerMetrics.size + 'px';
+        this.waveContainer.style.height = this.containerMetrics.size + 'px';
+      },
+
+      mouseUp: function (event) {
+        if (!this.isMouseDown) {
+          return;
+        }
+
+        this.mouseUpStart = Utility.now();
+      },
+
+      remove: function () {
+        this.waveContainer.remove();
+      }
+    };
+
+    Polymer({
+      is: 'paper-ripple',
+
+      published: {
+        /**
+         * The initial opacity set on the wave.
+         *
+         * @attribute initialOpacity
+         * @type number
+         * @default 0.25
+         */
+        initialOpacity: 0.25,
+
+        /**
+         * How fast (opacity per second) the wave fades out.
+         *
+         * @attribute opacityDecayVelocity
+         * @type number
+         * @default 0.8
+         */
+        opacityDecayVelocity: 0.8, 
+      },
+
+      configure: function () {
+        return {
+          initialOpacity: 0.25,
+          opacityDecayVelocity: 0.8,
+          ripples: [],
+          animating: false,
+          boundAnimate: this.animate.bind(this)
+        };
+      },
+
+      listeners: {
+        mousedown: 'onMouseDown',
+        mouseup: 'onMouseUp'
+      },
 
-          // Draw the ripple.
-          drawRipple(wave, x, y, radius, waveAlpha, bgFillAlpha);
-
-          // Determine whether there is any more rendering to be done.
-          var maximumWave = waveAtMaximum(wave, radius, anim);
-          var waveDissipated = waveDidFinish(wave, radius, anim);
-          var shouldKeepWave = !waveDissipated || maximumWave;
-          // keep rendering dissipating wave when at maximum radius on upAction
-          var shouldRenderWaveAgain = wave.mouseUpStart ? !waveDissipated : !maximumWave;
-          shouldRenderNextFrame = shouldRenderNextFrame || shouldRenderWaveAgain;
-          if (!shouldKeepWave || this.cancelled) {
-            deleteTheseWaves.push(wave);
+      get shouldKeepAnimating () {
+        for (var index = 0; index < this.ripples.length; ++index) {
+          if (!this.ripples[index].isAnimationComplete) {
+            return true;
           }
-       }
+        }
+
+        return false;
+      },
+
+      onMouseDown: function (event) {
+        var ripple = this.addRipple();
+
+        ripple.mouseDown(event);
+
+        if (!this.animating) {
+          this.animate();
+        }
+      },
+
+      onMouseUp: function (event) {
+        this.ripples.forEach(function (ripple) {
+          ripple.mouseUp(event);
+        });
+
+        this.animate();
+      },
+
+      onAnimationComplete: function () {
+        this.animating = false;
+        this.$.background.style.backgroundColor = null;
```

Comment:
Yup, there were bigger complexities on my mind while porting this. I'm just glad it all works :P